server/src/main/java/org/opensearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -69,7 +70,7 @@ public SimulatePipelineTransportAction(
             (Writeable.Reader<SimulatePipelineRequest>) SimulatePipelineRequest::new
         );
         this.ingestService = ingestService;
-        this.executionService = new SimulateExecutionService(threadPool);
+        this.executionService = new SimulateExecutionService(threadPool, clusterService);
```

Comment:
We can move validation for simulate API from here to `SimulatePipelineTransportAction`. There we have ref of `IngestService`, we can call our validation method from there.

https://github.com/opensearch-project/OpenSearch/blob/23cba282bbb3276781a71ed3ace315abb90f07d2/server/src/main/java/org/opensearch/action/ingest/SimulatePipelineTransportAction.java#L76-L91

Reply:
Will update this